kie-wb-common-react/boxed-expression-editor/src/components/LiteralExpression/LiteralExpression.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "./LiteralExpression.css";
+import * as React from "react";
+import { useCallback, useState } from "react";
+import { ExpressionProps, LiteralExpressionProps } from "../../api";
+import { TextArea } from "@patternfly/react-core";
+import { EditExpressionMenu } from "../EditExpressionMenu";
+
+export const LiteralExpression: React.FunctionComponent<LiteralExpressionProps> = ({
+  content,
+  dataType,
+  name,
+  updateNameAndDataTypeCallback,
+}: LiteralExpressionProps) => {
+  const [expressionName, setExpressionName] = useState(name);
+  const [expressionDataType, setExpressionDataType] = useState(dataType);
+  const [literalExpressionContent, setLiteralExpressionContent] = useState(content);
```

Comment:
Just FYI, if these props change, the state is not going to be updated, beware of that!

Reply:
Yeah, I know props purpose. Thanks.
I am retrieving props value as initial state. Is there any wrong usage with that?

Reply:
No, there's nothing wrong with that. I just mentioned it here because I struggled a little bit with this default state values at first :) Your usage is correct and if you understand that the state won't change when the props change without having a special useEffect for that, that's great!